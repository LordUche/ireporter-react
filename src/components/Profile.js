import React from 'react';
import { Link } from 'react-router-dom';
import { Button } from 'semantic-ui-react';

const Profile = () => {
  return (
    <div>
      <Button color="red" as={Link} to="/report" content="REPORT AN INCIDENT" />
    </div>
  );
};

export default Profile;
