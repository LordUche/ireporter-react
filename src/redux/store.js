import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { logger } from 'redux-logger';
import thunk from 'redux-thunk';
import rootReducer from './reducers';

const middleware = [thunk];

process.env.NODE_ENV === `development` && middleware.push(logger);

const initialState = { auth: { loggedIn: false, loading: false } };
const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(applyMiddleware(...middleware))
);

export default store;
