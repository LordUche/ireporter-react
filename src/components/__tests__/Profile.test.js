import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Provider } from 'react-redux';
import Profile from '../Profile';
import store from '../../redux/store';
import mockRouterOptions from '../../../test/__mocks__/mockRouter';

describe('<Profile />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <Profile />
        </Router>
      </Provider>,
      mockRouterOptions
    );
  });

  it('should match snapshot', () => {
    const tree = toJSON(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
