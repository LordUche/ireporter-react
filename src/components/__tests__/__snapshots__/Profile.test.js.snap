// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> should match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "pvmwqu",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "pvmwqu",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Profile>
        <div>
          <Button
            as={[Function]}
            color="red"
            content="REPORT AN INCIDENT"
            to="/report"
          >
            <Ref
              innerRef={
                Object {
                  "current": <a
                    class="ui red button"
                    href="/report"
                    role="button"
                  >
                    REPORT AN INCIDENT
                  </a>,
                }
              }
            >
              <RefFindNode
                innerRef={
                  Object {
                    "current": <a
                      class="ui red button"
                      href="/report"
                      role="button"
                    >
                      REPORT AN INCIDENT
                    </a>,
                  }
                }
              >
                <Link
                  className="ui red button"
                  onClick={[Function]}
                  role="button"
                  to="/report"
                >
                  <a
                    className="ui red button"
                    href="/report"
                    onClick={[Function]}
                    role="button"
                  >
                    REPORT AN INCIDENT
                  </a>
                </Link>
              </RefFindNode>
            </Ref>
          </Button>
        </div>
      </Profile>
    </Router>
  </MemoryRouter>
</Provider>
`;
